src/components/UserContext/UserContext.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react';
+import { User } from '../../types/User';
+import { getUsers } from '../../api/users';
+import { Post } from '../../types/Post';
+import { Comment } from '../../types/Comment';
+import { Errors } from '../../types/Errors';
+
+// at the end, move this props to a new folder
+type ContextProps = {
+  users: User[],
+  posts: Post[] | undefined,
+  setPosts: React.Dispatch<React.SetStateAction<Post[] | undefined>>,
+  choosedPost: Post | undefined,
+  setChoosedPost: React.Dispatch<React.SetStateAction<Post | undefined>>,
+  comments: Comment[] | undefined,
+  setComments: React.Dispatch<React.SetStateAction<Comment[] | undefined>>,
+  error: Errors,
+  setError: React.Dispatch<React.SetStateAction<Errors>>,
+  isLoading: boolean,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
+  isActiveComForm: boolean,
+  setIsActiveComForm: React.Dispatch<React.SetStateAction<boolean>>,
+};
+
+export const UserContext = React.createContext<ContextProps>({
```

Comment:
Not sure it's the correct name. It stores a lot of things not only those which are related directly to `user`

Moreover, it not store info about `user`